pkg/resources/view_grant.go

```diff
@@ -107,14 +110,23 @@ func CreateViewGrant(data *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	// ID format is <db_name>|<schema_name>|<view_name>|<privilege>
-	// view_name is empty when on_future = true
-	if futureViews {
-		data.SetId(fmt.Sprintf("%v|%v||%v", dbName, schemaName, priv))
-	} else {
-		data.SetId(fmt.Sprintf("%v|%v|%v|%v", dbName, schemaName, viewName, priv))
+	data_identifiers := [][]string{{dbName, schemaName, viewName, ""}}
+
+	if !(futureViews) { // merge with above if/else statement if possible
+		data_identifiers[0][3] = priv
 	}
 
+	var buf bytes.Buffer
+	csv_writer := csv.NewWriter(&buf)
```

Comment:
in general, use camelCase for Go
```suggestion
	csvWriter := csv.NewWriter(&buf)
```